### PR TITLE
nfs: only set the pool size when it exists and always run default pool creation

### LINKED ISSUE #9205: nfs-ganesha pool does not have application type set
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Upon upgrading from rook 1.6.8 -> 1.7.7 a new pool named `nfs-ganesha` was created in my cluster without an application type set, which causes a health warning:

image

```bash
[root@rook-ceph-tools-ff5dc774-stjl5 /]# ceph status
  cluster:
    id:     51f94ef6-6ac8-4853-9dde-a2f2ec077ff7
    health: HEALTH_WARN
            1 pool(s) do not have an application enabled
            1 daemons have recently crashed
 
  services:
    mon: 3 daemons, quorum a,b,d (age 23m)
    mgr: b(active, since 22m), standbys: a
    mds: 12/12 daemons up, 12 hot standby
    osd: 24 osds: 24 up (since 18m), 24 in (since 16h)
 
  data:
    volumes: 4/4 healthy
    pools:   11 pools, 417 pgs
    objects: 251.72k objects, 953 GiB
    usage:   2.8 TiB used, 7.7 TiB / 10 TiB avail
    pgs:     417 active+clean
 
  io:
    client:   14 KiB/s rd, 793 KiB/s wr, 26 op/s rd, 51 op/s wr
```

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

For CRD not using the new nfs spec that includes the pool settings,
applying the "size" property won't work since it is set to 0. The pool
still gets created but returns an error. The loop is re-queued but on
the second run the pool is detected so no further configuration is done.

 Previously, if the pool was present we would not run the pool creation
again. This is a problem if the pool spec changes, the new settings will
never be applied.

Closes: https://github.com/rook/rook/issues/9205
Signed-off-by: Sébastien Han <seb@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves #9205

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/nfs/controller_test.go [hunk 1]
```diff
@@ -19,10 +19,11 @@ package nfs
 
 import (
 	"context"
-	"errors"
 	"os"
 	"testing"
 
+	"github.com/pkg/errors"
+
```
Comment (BlaineEXE) [inline pkg/operator/ceph/nfs/controller_test.go:26]: Do we want this newline?

**Comment by BlaineEXE** [comment]
There may also be an issue where we aren't setting the `application` on the Pool to `nfs` when creating the default pool. I have been meaning to look more closely at that after chatting with someone on slack. Have you observed this?

**Comment by leseb** [comment]
> There may also be an issue where we aren't setting the `application` on the Pool to `nfs` when creating the default pool. I have been meaning to look more closely at that after chatting with someone on slack. Have you observed this?

This is actually a side effect of this bug.

**Review by travisn — APPROVED** [review]

**Comment by BlaineEXE** [comment]
> > There may also be an issue where we aren't setting the `application` on the Pool to `nfs` when creating the default pool. I have been meaning to look more closely at that after chatting with someone on slack. Have you observed this?
> 
> This is actually a side effect of this bug.

Good to hear :)

**Review by BlaineEXE — APPROVED** [review]
Going to approve, maybe Seb will want to remove the newline and then merge. :)

**Comment by leseb** [comment]
> Going to approve, maybe Seb will want to remove the newline and then merge. :)

Done.

**Pull request merged**